src/motor/motor_ip.cpp

```diff
@@ -0,0 +1,149 @@
+#include "motor_ip.h"
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include <errno.h>
+#include "poll.h"
+
+namespace obot {
+
+void UDPFile::open() {
+    fd_ = ::socket(AF_INET, SOCK_DGRAM, 0);
+    addrinfo hints = {};
+    hints.ai_family = AF_INET;      // IPv4
+    hints.ai_socktype = SOCK_DGRAM; // UDP
+
+    addrinfo *result, *res;
+    int addr_info_result = getaddrinfo(ip_.c_str(), port_.c_str(), &hints, &result);
+    if (addr_info_result != 0) {
+      throw std::runtime_error("getaddrinfo failed for " + ip_ + ":" + port_ + ", error: " + std::to_string(addr_info_result));
+    }
+
+    char addrstr[100];
+    void *ptr;
+    res = result;
+    int n_results = 0;
+    while(res) {
+        n_results++;
+        socklen_t addr_len = 0;
+        if (res->ai_family != AF_INET) {
+          throw std::runtime_error(ip_ + ":" + port_ + ", not ipv4");
+        }
+        ptr = &((struct sockaddr_in *) res->ai_addr)->sin_addr;
+        addr_len = sizeof(sockaddr_in);
+        inet_ntop (res->ai_family, ptr, addrstr, 100);
+        addrstr_ = addrstr;
+        res = res->ai_next;
+    }
+    if (n_results != 1) {
+      throw std::runtime_error(ip_ + ":" + port_ + ", n_results error: " + std::to_string(n_results));
+    }
+    std::memcpy(&addr_, result->ai_addr, result->ai_addrlen);
+    freeaddrinfo(result);
+    flush();
+}
+
+void UDPFile::flush() {
+    int result;
+    char c[1024];
```

Comment:
Super minor but if this where a std::array<char, 1024> then below you could do read(c.data(), c.size()) which can help make sure that one is never changed without the other leading to buffer overrun. 

Also maybe not a problem but this is a fairly large chunk to allocate of the stack. Probably ok but stack overflows are annoyingly hard to trace down sometimes.